src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -281,10 +281,50 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     return this._loadPromise;
   }
 
+  /**
+   * handle decode error - reload the video and seek to last currentTime
+   * @returns {void}
+   */
+  handleDecodeError(): void {
+    NativeAdapter._logger.debug(`handleDecodeError currentTime <${this._videoElement.currentTime}>`);
+    this._playerTracks = this._getParsedTracks();
+    const prevCurrTime = this._videoElement.currentTime;
+    const prevActiveAudioTrack = this._getActivePKAudioTrack();
+    const prevActiveTextTrack = this._getActivePKTextTrack();
+
+    this._eventManager.removeAll();
+    this._loadPromise = null;
+
+    if (this._videoElement && this._videoElement.src) {
+      Utils.Dom.setAttribute(this._videoElement, 'src', '');
+      Utils.Dom.removeAttribute(this._videoElement, 'src');
+    }
+
+    this.load(prevCurrTime).then(() => {
+      this._eventManager.listenOnce(this._videoElement, Html5EventType.PLAYING, () => {
+        if (prevActiveAudioTrack) {
+          this.selectAudioTrack(prevActiveAudioTrack);
+        }
+        if (prevActiveTextTrack) {
+          this.selectTextTrack(prevActiveTextTrack);
+        } else {
+          this._disableTextTracks();
+        }
+      });
+
+      this._videoElement.play();
+    });
+  }
+
   handleMediaError(error: ?MediaError): boolean {
     if (this._loadPromiseReject) {
+      NativeAdapter._logger.debug('handleMediaError _loadPromiseReject');
       this._loadPromiseReject(new Error(Error.Severity.CRITICAL, Error.Category.MEDIA, Error.Code.NATIVE_ADAPTER_LOAD_FAILED, error));
       return true;
+      // It seems that the only known scenario to reach this code with a decode error (3) is as described in FEC-8873
+    } else if (error && error.code === 3) {
```

Comment:
need to add retry params so we don't try to recover and fail forever.
Use platform ENUM instead of 3, i.e. `MediaError.MEDIA_ERR_DECODE`, see https://developer.mozilla.org/en-US/docs/Web/API/MediaError